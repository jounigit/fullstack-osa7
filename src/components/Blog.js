import React from 'react'

class Blog extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      visible: false
    }
  }

  toggleVisibility = () => {
    this.setState({ visible: !this.state.visible })
  }

  render() {
    //const hideWhenVisible = { display: this.state.visible ? 'none' : '' }
    const showWhenVisible = {
      display: this.state.visible ? '' : 'none',
      paddingLeft: 12
     }

    const blogStyle = {
      paddingTop: 10,
      paddingLeft: 2,
      border: 'solid',
      borderWidth: 1,
      marginBottom: 5
    }

    return (
      <div style={blogStyle}>
        <div onClick={this.toggleVisibility} className="titleAndAuthor">
          <div className="title" style={{display: 'inline'}}>{this.props.title},</div>
          <div className="author" style={{display: 'inline'}}>{this.props.author}</div>
        </div>
        <div style={showWhenVisible} className="content">
            <a href="{this.props.url}">{this.props.url}</a><br />
            {this.props.likes} likes <button onClick={this.props.toggleLike}>like</button><br />
            added by {this.props.name}<br />
            {this.props.name === this.props.loggedUser || this.props.name === 'anonymous' ?
              <button onClick={this.props.moveBlog}>delete</button> : ''
            }
        </div>
      </div>
    )
  }
}

export default Blog
