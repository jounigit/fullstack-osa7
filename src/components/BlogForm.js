import React from 'react'

const BlogForm = ({handleSubmit, handleChange, blogTitle, blogAuthor, blogUrl}) => {
  return (
    <div>
      <h2>create new</h2>

      <form onSubmit={handleSubmit}>
        <div>
          Title
          <input
            type="text"
            name="blogTitle"
            value={blogTitle}
            onChange={handleChange}
          />
        </div>
        <div>
          Author
          <input
            type="text"
            name="blogAuthor"
            value={blogAuthor}
            onChange={handleChange}
          />
        </div>
        <div>
          Url
          <input
            type="text"
            name="blogUrl"
            value={blogUrl}
            onChange={handleChange}
          />
        </div>
        <button type="submit">tallenna</button>
      </form>
    </div>
  )
}

export default BlogForm
