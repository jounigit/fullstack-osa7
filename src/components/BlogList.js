import React from 'react'
import { connect } from 'react-redux'
import {  Link } from 'react-router-dom'
import { blogsInit } from '../reducers/blogReducer'

class BlogList extends React.Component {
  componentDidMount = async () => {
    this.props.blogsInit()
  }
  render() {
    const blogStyle = {
      paddingTop: 10,
      paddingLeft: 5,
      paddingBottom: 5,
      border: 'solid',
      borderWidth: 1,
      marginBottom: 5
    }
    console.log('BLOGLIST: ', this.props.blogs)
    if (this.props.blogs.length !== 0) {
    return (
      <div>
        <h2>blogs</h2>
        {this.props.blogs.map(blog =>
          <div key={blog.id}  style={blogStyle}>

            <Link to={`/blogs/${blog.id}`}>
              <div className="titleAndAuthor">
                {blog.title} {blog.author}
              </div>
            </Link>

          </div>
        )}
      </div>
    )
  }

  return (
    <div>
      <h2>waiting</h2>

      </div>
  )

}

}

const mapStateToProps = (state) => {
  return {
    blogs: state.blogs
  }
}

const mapDispatchToProps = {
  //notify,
  blogsInit
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(BlogList)
