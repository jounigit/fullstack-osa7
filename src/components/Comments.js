import React from 'react'
import { connect } from 'react-redux'
import { notify } from '../reducers/notificationReducer'
import blogService from '../services/blogs'

class Comments extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      comments: [],
      blog: null
    }
  }

  componentDidMount = async () => {
    const { blog } = this.props
    this.setState({
      comments: blog.comments,
      blog: blog
    })
  }

  handleSubmit = async (e) => {
    e.preventDefault()
    try {
      const content = e.target.comment.value
      e.target.comment.value = ''

      const commentObj = {
        comment: content
      }

      await blogService.addComment(this.state.blog.id, commentObj)

      this.setState({
        comments: this.state.comments.concat(content)
      })
      this.props.notify(`your comment '${content}'`, 10)
    } catch (exception) {
      this.props.notify('updating error', 10, 'error')
    }
  }

  render() {
    const comments = this.state.comments.map((c, index) =>
      <li key={index}>
        {c}
      </li>
    )
    return (
      <div>
        <h3>comments</h3>

        <form onSubmit={this.handleSubmit}  style={{ display: 'inline !important' }}>
          <div><input name='comment'/></div>
          <button>add comment</button>
        </form>

        <ul>
          {comments}
        </ul>
      </div>
    )
  }
}

export default connect(
  null,
  { notify }
)(Comments)
