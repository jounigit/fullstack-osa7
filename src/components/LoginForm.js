import React from 'react'
import { connect } from 'react-redux'
import { login } from '../reducers/loginReducer'
import { Form, Button, Header, Card } from 'semantic-ui-react'
//import PropTypes from 'prop-types'

export class LoginForm extends React.Component {
  handleSubmit = async (e) => {
    e.preventDefault()

    this.props.login(e.target.username.value, e.target.password.value)
    e.target.username.value = ''
    e.target.password.value = ''
  }
  render () {
    return (
      <Card centered style={{ marginTop: 20 }}>
        <Card.Content>
          <Header as='h2' color='green'>Kirjaudu sovellukseen</Header>
        </Card.Content>
        <Card.Content>
          <Form onSubmit={this.handleSubmit}>
            <Form.Field>
              <label>käyttäjätunnus</label>
              <input type="text" name="username" />
            </Form.Field>
            <Form.Field>
              <label>salasana</label>
              <input type="password" name="password" />
            </Form.Field>
            <Button basic color='green'>kirjaudu</Button>
          </Form>
        </Card.Content>
      </Card>
    )
  }
}
/*
LoginForm.propTypes = {
  username: PropTypes.string,
  password: PropTypes.string
} */

export default connect(
  null,
  { login }
)(LoginForm)
