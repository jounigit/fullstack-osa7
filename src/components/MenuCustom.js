import React from 'react'
import { NavLink } from 'react-router-dom'
import { Button } from 'semantic-ui-react'

const MenuCustom = ({ user, logout }) => (
  <div className="menu">
    <NavLink
      exact activeStyle={{
        fontWeight: 'bold',
        color: 'DarkRed'
      }} to="/">BLOGS</NavLink> &nbsp;
    <NavLink
      exact activeStyle={{
        fontWeight: 'bold',
        color: 'DarkRed'
      }} to="/users">USERS</NavLink> &nbsp;
    <em>{user.name} logged in</em> &nbsp;
    {user && <Button size='tiny' color='teal' onClick={logout}>logout</Button>}
  </div>
)

export default MenuCustom
