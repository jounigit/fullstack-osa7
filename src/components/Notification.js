import React from 'react'
import { connect } from 'react-redux'
import '../App.css'

class Notification extends React.Component {
  render() {
    const { notification } = this.props
    const visibility = { display: notification.visibility ? '' : 'none' }
    return (
      <div style={visibility}>
        <div className={ notification.style }>
          { notification.content }
        </div>
      </div>
    )
  }
}

const mapStateToProps = (state) => {
  return {
    notification: state.notification
  }
}

export default connect(
  mapStateToProps
)(Notification)
