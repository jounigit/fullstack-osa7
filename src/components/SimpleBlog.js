import React from 'react'

const SimpleBlog = ({ blog, onClick }) => (
  <div>
    <div>
      <div className="title" style={{display: 'inline'}}>{blog.title},</div>
      <div className="author" style={{display: 'inline'}}>{blog.author}</div>
    </div>
    <div className="likes">
      blog has {blog.likes} likes
      <button onClick={onClick}>like</button>
    </div>
  </div>
)

export default SimpleBlog
