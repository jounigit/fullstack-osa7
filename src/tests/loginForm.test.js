import React from 'react'
import { LoginForm } from '../components/LoginForm'
import renderer from 'react-test-renderer'

it('+++ LOGINFORM renders correctly', () => {
  const tree = renderer
    .create(<LoginForm username='tollo' password='passi' />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

/*
inputA = wrapper.findAll('input').at(0)
inputA.setValue('123456')
*/
