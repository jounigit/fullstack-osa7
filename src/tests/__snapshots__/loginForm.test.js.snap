// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`+++ LOGINFORM renders correctly 1`] = `
<div
  className="ui centered card"
  href={undefined}
  onClick={[Function]}
  style={
    Object {
      "marginTop": 20,
    }
  }
>
  <div
    className="content"
  >
    <h2
      className="ui green header"
    >
      Kirjaudu sovellukseen
    </h2>
  </div>
  <div
    className="content"
  >
    <form
      action={undefined}
      className="ui form"
      onSubmit={[Function]}
    >
      <div
        className="field"
      >
        <label>
          käyttäjätunnus
        </label>
        <input
          name="username"
          type="text"
        />
      </div>
      <div
        className="field"
      >
        <label>
          salasana
        </label>
        <input
          name="password"
          type="password"
        />
      </div>
      <button
        className="ui green basic button"
        disabled={undefined}
        onClick={[Function]}
        role="button"
        tabIndex={undefined}
      >
        kirjaudu
      </button>
    </form>
  </div>
</div>
`;
